packages/blocks/src/store/reducer.js

```diff
@@ -372,19 +372,17 @@ export function collections( state = {}, action ) {
 }
 
 export function blockBindingsSources( state = {}, action ) {
+	// Merge usesContext with existing values, potentially defined in the server registration.
+	const existingUsesContext = state[ action.name ]?.usesContext || [];
```

Comment:
I had one realization afte this PR got merged. Currently, all reducers get executed whenever any action gets dispatched. In effect, the logic added on top of the function will get executed each time. I would recommend turning it into an util that gets called inside the `switch` in case it is really needed. 

The following could get added as an util:

```js
/**
 * Merges usesContext with existing values, potentially defined in the server registration.
 *
 * @param {string[]} existingUsesContext Existing `usesContext`.
 * @param {string[]} newUsesContext       Newly added `usesContext`.
 * @return {string[]|undefined} Merged `usesContext`.
 */
function getMergedUsesContext( existingUsesContext = [],  newUsesContext = [] ) {
	const mergedArrays = Array.from(
		new Set( existingUsesContext.concat( newUsesContext ) )
	);
	return mergedArrays.length > 0 ? mergedArrays : undefined;
}
```

That could be then used when updating reducers.

Reply:
I have created a follow-up pull request to discuss it and potentially fix it: [link](https://github.com/WordPress/gutenberg/pull/63941).